climate-monitoring-client/src/main/java/climatemonitoring/ProxyImpl.java

```diff
@@ -628,33 +698,37 @@ public synchronized Parameter[] getParameters(int geoname_id, String center_id)
 		} catch (ClassNotFoundException e){
 			e.printStackTrace();
 		}
-		
+
 		return getparameters;
 	}
-	
+
 	/**
-	 * Returns an array containing parameters about a specified area from
-	 * the last center that submitted a parameter
+	 * To get the average about the score of a specific area,
+	 * of a specific center about a specific category
 	 * 
 	 * @param geoname_id The area's ID
-	 * @return The result of the search as an array of parameters
+	 * @param center_id The center's ID
+	 * @param category The parameter's category
+	 * @return The average of the scores as a double
 	 * @throws ConnectionLostException If the client loses connection during the operation
 	 * @throws DatabaseRequestException If the database fails to process the given request
 	 */
 	@Override
-	public Parameter[] getParameters(int geoname_id) throws ConnectionLostException, DatabaseRequestException {
+	public double getParametersAverage(int geoname_id, String center_id, String category) throws ConnectionLostException, DatabaseRequestException {
 
-		Parameter[] getparameters = null;
+		double getparametersaverage = 0.0;
 
 		try {
 
-			out.writeObject(RequestType.GET_PARAMETERS_AREA);
+			out.writeObject(RequestType.GET_PARAMETERS_AVERAGE);
 			out.writeObject(geoname_id);
+			out.writeObject(center_id);
+			out.writeObject(category);
 
 			boolean success = (boolean) in.readObject();
 
 			if(success == true){
-				getparameters = (Parameter[]) in.readObject();
+				getparametersaverage = (double) in.readObject();
 			}else{
 				DatabaseRequestException e = (DatabaseRequestException) in.readObject();
```

Comment:
The casting should be done with the `Double` class instead of the primitive type `double`